src/ligero/prover.rs

```diff
@@ -20,29 +22,20 @@ const MAX_RUN_LENGTH: usize = 1 << 25;
 
 /// Prover for Ligero.
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub struct LigeroProver<'a> {
-    tableau_layout: &'a TableauLayout<'a>,
-}
-
-impl<'a> LigeroProver<'a> {
-    /// Construct a new prover from the Ligero parameters.
-    pub fn new(tableau_layout: &'a TableauLayout<'a>) -> Self {
-        Self { tableau_layout }
-    }
+pub struct LigeroProver {}
```

Comment:
This struct is now useless, and I will remove it and reduce `ligero_prove` to a bare function. I didn't do it in this PR because then the entire function would appear in the diff because it got de-indented by one step.